x-pack/plugins/ml/public/application/model_management/add_model_flyout.tsx

```diff
@@ -0,0 +1,445 @@
+/*
```

Comment:
I believe there was discussion about other plugins being able to make use of this flyout?
If so I think we should create a new uiAction to provide it. It would also mean the flyout itself should contain the ability to trigger the download rather than having it supplied via a callback.

Reply:
Do we have an issue to track it? We can also share it from the plugin `start` if needed. 
I reckon it deserves a dedicated PR.

Reply:
I'm thinking of the design doc which was created by the search team. I do not know of an issue which covers the work.
Yeah, making these changes in a dedicated PR makes sense when the work is needed.